### ISSUE TITLE
LIS does not create the required DA sub-directories with WMO convention output

### ISSUE THREAD

**Issue by dmocko** [NASA-LIS/LISF#383]
When "WMO convention" is chosen in the lis.config file:

Output naming style:                    "WMO convention"

...LIS can crash when trying to write DA (data assimilation) and/or perturbation file output/restart files.

The issue is that LIS assumes that the datasets will be written into separate sub-directories, depending on the type of data being written.  However, only when using "WMO convention", LIS does not actually create these sub-directories before it later tries to write data into them.

This issue is also referenced in Trac ticket number 630.  (Trac is restricted only to those of us at NASA).  Further discussion of this Issue will take place only here on Github.

**Comment by dmocko** [comment]
Looking into this more, and coming up with a solution, I have found when using "WMO convention" output naming style:

1) SURFACEMODEL output files are written in the LIS "Output directory:" set in lis.config.

   I believe we want to keep this as is for "WMO convention".

2) DAPERT output files are written in a "DAPERT" sub-directory off the "Output directory:".

   I have a solution to ensure LIS creates this directory for "WMO convention"; however, should we instead change the directory into which this data is written?  This would not be difficult to change for "WMO convention".

3) DAOBS output files are written into "DAOBS/[YYYYMM]" sub-directories.

   I have a solution to ensure LIS creates these directories for "WMO convention"; however, to make the same change as proposed in "2)" above will require also changing a subroutine in every directory in "dataassim/obs".  It's about 40 files that need to be changed, if we change the directory into which DAOBS data is output.

4) EnKF output files are written directory into the LIS "Output directory:" when using "WMO convention".

   No solution needed, as this directory is also created.  But asking (the inverse of) the same question - should we change the default directory for "WMO convention" to instead write the data into a "EnKF" sub-directory?

To summarize, when we choose "WMO convention" output style, should:

A) All data be written into the LIS "Output directory:"?

or

B) SURFACEMODEL output be written into the LIS "Output directory:" and DAPERT, DAOBS, and EnKF data be written into separate sub-directories?

or

C) Leave the data written where currently is in the code, and only ensure that those directories are there before LIS tries to write the data?

**Comment by dmocko** [comment]
For now, just going with option "C", which is to leave the data written into the directories as the code is now.  The only fix is to ensure that any relevant directories are created by LIS before other parts of LIS tries to write data into those directories.

A pull request is coming, but we can consider re-opening this issue at a later date if we wanted to modify other parts of LIS to change the sub-directories that "WMO convention" output style will use.

**Comment by emkemp** [comment]
Pull request has been merged, so I will close this issue.
